### PR TITLE
JENKINS: Set correct C++ compiler when using Intel compiler.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

File: contrib/test_jenkins.sh [hunk 1]
```diff
@@ -182,7 +182,7 @@ if [ -n "$JENKINS_RUN_TESTS" ]; then
     module unload hpcx-gcc
 
     module load intel/ics
-    ../contrib/configure-devel --prefix=$ucx_inst CC=icc
+    ../contrib/configure-devel --prefix=$ucx_inst CC=icc CXX=icc
```
Comment (shssf) [inline contrib/test_jenkins.sh]: Intel C++ compiler is `icpc`

**Comment by shssf** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3322/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/853/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3323/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/854/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2813/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2814/ for details (Mellanox internal link).

**Pull request merged**